components/table/nz-table.component.ts

```diff
@@ -307,4 +322,26 @@ export class NzTableComponent<T = any> implements OnInit, AfterViewInit, OnDestr
     this.destroy$.next();
     this.destroy$.complete();
   }
+
+  private handleDataChange(nzData?: T[] | NzDataSource<T>): void {
+    if (isDataSource(nzData)) {
+      this.isIteratorOverriddenOrThrow(nzData);
+      this.data = nzData;
+    } else {
+      this.data = new NzArrayDataSource(nzData || [], !this.nzFrontPagination);
+    }
+
+    if (this.nzFrontPagination) {
+      this.data = new NzDetachedDataSource(this, this.data);
+      this.data.connect(this).subscribe({
+        next: data => this.nzCurrentPageDataChange.emit(data as T[])
+      });
+    }
+  }
+
+  private isIteratorOverriddenOrThrow(nzData: NzDataSource<T>): void {
```

Comment:
This should return a `boolean`, and should be called after `isDataSource(nzData)`